### ISSUE TITLE
i18n/translate templates feature?

### ISSUE THREAD

**Issue by p2227** [esdoc/esdoc#319]
i want to translate some words in template

for example
https://github.com/esdoc/esdoc/blob/e70951329893047d79264ceaa0f68c4860de8c16/src/Publisher/Builder/template/properties.html#L5

how can i do this ? develop a plugin ?

or providing a config named `template` in `esdoc.json` ?

**Comment by typhonrt** [comment]
Hi @p2227, welcome to ESDoc though a belated in response! This situation is tricky presently and there is no shortcut per se. A plugin won't exactly do the job as the only thing that could really be modified is the HTML output of the default static HTML publisher in `onHandleHTML`. Parsing / modifying HTML / text directly can be cumbersome for i18n purposes.

ESDoc has a built in [publisher](https://github.com/esdoc/esdoc/tree/master/src/Publisher) which produces the static HTML documentation. At this time the only way to swap an alternate publisher component is to programmatically invoke ESDoc; see the [publisher section](https://esdoc.org/api.html) of the API docs. Conceivably an entirely alternate publisher component could be provided programmatically. You would basically have to duplicate the functionality found under the `Publisher` directory. Doing this allows the [templates](https://github.com/esdoc/esdoc/tree/master/src/Publisher/Builder/template) to be fully controlled and i18n functionality to be potentially added.

Ideally in the near future it would be great for a parameter such as `publisher` to be indexed in the ESDoc configuration file which would load an alternate NPM module for publishing (similar to the plugin API).

Another possible future enhancement to ESDoc is to include i18n concerns into the default publisher and have a plugin callback which could translate marked text in templates. 

Basically it's possible to work out a proof of concept solution now, but with future enhancements could be made a bit easier without having to track any / all changes that occur in the default publisher code.

**Comment by p2227** [comment]
thanks @typhonrt  reply.
A better way is providing a EJS/mustache like template .Not to programming. But as you said, there is no shortcut per se now.

**Comment by h13i32maru** [comment]
@p2227 ESDoc does not support i18n. Please use [Plugin API](https://esdoc.org/manual/advanced/api.html#plugin-feature)

A rough code:
`i18n-plugin.js`
```js
const cheerio = require('cheerio');

const transalteMap = {
  'Static Public Function Summary': 'your-trnaslated-text'
}

exports.onHandleHTML = function(ev) {
  if (!ev.data.fileName.includes('.html')) return;

  const $ = cheerio.load(ev.data.html);

  function traverse($el) {
    const children = $el.get(0).children;
    for (const el of children) {
      switch (el.type) {
      case 'text':
        if (transalteMap[el.data]) {
          el.data = transalteMap[el.data];
        }
        break;
      case 'tag':
        traverse($(el));
        break;
      }
    }
  }

  traverse($('body'));

  ev.data.html = $.root().html();
};
```

`esdoc.json`
```json
{
  "source": "./src",
  "destination": "./doc",
  "plugins": [
    {"name": "./i18n-plugin.js"}
  ]
}
```
